### PR TITLE
Add support for open_as_secondary APIs.

### PR DESCRIPTION
**Pull request by calavera**
This change adds support for the OpenAsSecondary APIs to open a database
as a secondary. It also adds support for TryCatchUpWithPrimary to allow
clients to synchronize primary and secondary.

/cc @aleksuss 

Signed-off-by: David Calavera <david.calavera@gmail.com>

### CODE REVIEW

**Comment by aleksuss** [comment]
Overall looks good. But to be honest I'd like to see some tests for covering this functionality. Could you add a couple ? Thanks.

File: tests/test_db.rs [hunk 1]
```diff
@@ -337,3 +337,28 @@ fn test_get_updates_since_out_of_range() {
...
+
+    let mut opts = Options::default();
+    opts.set_max_open_files(-1);
+
+    let secondary_path = DBPath::new("_rust_rocksdb_test_open_as_secondary_secondary");
+    let secondary = DB::open_as_secondary(&opts, &primary_path, &secondary_path).unwrap();
+
+    let result = secondary.get(b"key1");
+    let vector = result.unwrap().unwrap();
+    assert!(get_byte_slice(&vector) == b"value1");
```
Comment (aleksuss) [inline tests/test_db.rs:356]:
```suggestion
    assert_eq!(get_byte_slice(&vector), b"value1");
```

File: tests/test_db.rs [hunk 2]
```diff
@@ -337,3 +337,28 @@ fn test_get_updates_since_out_of_range() {
...
+
+    db.put(b"key1", b"value2").unwrap();
+    assert!(secondary.try_catch_up_with_primary().is_ok());
+
+    let result = secondary.get(b"key1");
+    let vector = result.unwrap().unwrap();
+    assert!(get_byte_slice(&vector) == b"value2");
```
Comment (aleksuss) [inline tests/test_db.rs:363]:
```suggestion
    assert_eq!(get_byte_slice(&vector), b"value2");
```

File: tests/test_db.rs [hunk 3]
```diff
@@ -337,3 +337,28 @@ fn test_get_updates_since_out_of_range() {
...
+    db.put(b"key1", b"value1").unwrap();
+
+    let mut opts = Options::default();
+    opts.set_max_open_files(-1);
+
+    let secondary_path = DBPath::new("_rust_rocksdb_test_open_as_secondary_secondary");
+    let secondary = DB::open_as_secondary(&opts, &primary_path, &secondary_path).unwrap();
+
+    let result = secondary.get(b"key1");
+    let vector = result.unwrap().unwrap();
```
Comment (aleksuss) [inline tests/test_db.rs:355]: This string seems to be redundant. I'd propose like:
```rust
let result = secondary.get(b"key1").unwrap().unwrap();
```

**Comment by calavera** [comment]
> Overall looks good. But to be honest I'd like to see some tests for covering this functionality. Could you add a couple ? Thanks.

I've added a test @aleksuss, thanks for the reminder! :raised_hands:

**Pull request merged**